artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyAcceptor.java

```diff
@@ -468,8 +473,12 @@ public synchronized SslHandler getSslHandler(ByteBufAllocator alloc) throws Exce
 
       engine.setUseClientMode(false);
 
-      if (needClientAuth)
+      if (needClientAuth) {
          engine.setNeedClientAuth(true);
+      }
+      if (wantClientAuth) {
```

Comment:
I would probably do this in an else as the setting would override the need option and if someone did something silly like specify both I'd generally prefer it chose the stronger of the two and setting want after need overrides it.